R/string_boundary_linter.R

```diff
@@ -0,0 +1,86 @@
+#' Require usage of startsWith() and endsWith() over grepl()/substr() versions
+#'
+#' [startsWith()] is used to detect fixed initial substrings; it is more
+#'   readable and more efficient than equivalents using [grepl()] or [substr()].
+#'   c.f. `startsWith(x, "abc")`, `grepl("^abc", x)`,
+#'   `substr(x, 1L, 3L) == "abc"`.
+#'
+#' Ditto for using [endsWith()] to detect fixed terminal substrings.
+#'
+#' @evalRd rd_tags("string_boundary_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+string_boundary_linter <- function() {
+  str_cond <- xp_and(
+    "string-length(text()) > 3",
+    "contains(text(), '^') or contains(text(), '$')"
+  )
+  grepl_xpath <- glue::glue("//expr[
+    expr[SYMBOL_FUNCTION_CALL[text() = 'grepl']]
+    and not(SYMBOL_SUB[
+      text() = 'ignore.case'
+      and not(following-sibling::expr[1][NUM_CONST[text() = 'FALSE']])
```

Comment:
If we check for passed values, might as well check for `F` too.

Reply:
oh right, we need `SYMBOL[text() = 'F']` for that... fixing...